carAPI/src/main/java/com/everis/control/PersistenceService.java

```diff
@@ -6,15 +6,56 @@
 import javax.persistence.EntityManager;
 import javax.persistence.PersistenceContext;
 import javax.persistence.TypedQuery;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+import javax.persistence.criteria.Predicate;
 
 @Stateless
 public class PersistenceService<T,I> {
 	
 	@PersistenceContext(unitName = "carAPI")
 	private EntityManager em;
 
-	public List<T> getAll(String genericQuery, Class<T> c){
-		TypedQuery<T> query = em.createNamedQuery(genericQuery, c);
+	public List<T> getAll(String genericQuery, Class<T> c, String filterBy, int pages, String orderBy, int size, String sort){
+		
+		if(sort==null) {
+			sort="asc";
+		}
+		if(orderBy==null) {
+			orderBy="id";
+		}
+		if(pages<=0) {
+			pages=0;
+		}
+		if(size<=0 || size >=20) {
+			size=10;
+		}
+		
+		CriteriaBuilder criteriaBuilder = em.getCriteriaBuilder();
+		CriteriaQuery<T> criteriaQuery = criteriaBuilder.createQuery(c);
```

Comment:
Long functions are difficult to read and understand, try to split the functionality, that way you could also reuse the code if you need it somewhere else. Also makes methods easier to test.